pennylane/tape/qscript.py

```diff
@@ -743,11 +749,18 @@ def _multi_homogenous_measurement_shape(mps, device):
 
             elif ret_type == qml.measurements.Sample:
 
-                shape = (len(mps), device.shots)
+                dim = mps[0].shape(device)
```

Comment:
Just to make sure: any difference in doing `mps[0].shape(device)` instead of `device.shots` as before? Good idea to change because `mps[0].shape(device)` should encapsulate this logic. :+1:

Reply:
I believe this is a bug. The original only works if `qml.sample()` is called with an observable. It won't work if `qml.sample()` is called with wires (or no arguments). This change makes it work in both cases.

Reply:
Amazing! Great catch.